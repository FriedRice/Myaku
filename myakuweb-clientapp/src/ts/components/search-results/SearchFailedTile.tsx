/**
 * Search failure notice tile component.
 * @module ts/components/search-results/SearchFailedTile
 */

import React from 'react';
import Tile from 'ts/components/generic/Tile';
import { useEffect } from 'react';


const SearchFailedTile: React.FC<{}> = function() {
    useEffect(function() {
        document.title = 'Search Failed';
    }, []);

    return (
        <div className='start-tile-container'>
            <Tile tileClasses='start-tile'>
                <h4 className='error-text'>Search Failed</h4>
                <p>
                    There was an issue connecting to the server to make the
                    search.
                </p>
                <p>Please attempt the search again later.</p>
            </Tile>
        </div>
    );
};

export default SearchFailedTile;
