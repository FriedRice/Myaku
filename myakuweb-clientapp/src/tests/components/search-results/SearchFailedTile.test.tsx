/**
 * Tests for the SearchFailedTile component.
 * @module tests/components/search-results/SearchFailedTile.test
 */

import React from 'react';
import SearchFailedTile from 'ts/components/search-results/SearchFailedTile';

import {
    mount,
    shallow,
} from 'enzyme';


describe('<SearchFailedTile />', function() {
    it('renders correctly', function() {
        const wrapper = shallow(<SearchFailedTile />);
        expect(wrapper).toMatchSnapshot();
    });

    it('sets document title', function() {
        mount(<SearchFailedTile />);
        expect(document.title).toBe('Search Failed');
    });
});
