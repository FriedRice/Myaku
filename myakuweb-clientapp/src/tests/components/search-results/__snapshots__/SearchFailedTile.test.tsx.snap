// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchFailedTile /> renders correctly 1`] = `
<div
  className="start-tile-container"
>
  <Tile
    tileClasses="start-tile"
  >
    <h4
      className="error-text"
    >
      Search Failed
    </h4>
    <p>
      There was an issue connecting to the server to make the search.
    </p>
    <p>
      Please attempt the search again later.
    </p>
  </Tile>
</div>
`;
